source/common/router/router.cc

```diff
@@ -226,13 +229,22 @@ FilterUtility::finalTimeout(const RouteEntry& route, Http::RequestHeaderMap& req
   return timeout;
 }
 
-bool FilterUtility::trySetGlobalTimeout(const Http::HeaderEntry* header_timeout_entry,
-                                        TimeoutData& timeout) {
+absl::optional<std::chrono::milliseconds>
+FilterUtility::tryParseHeaderTimeout(const Http::HeaderEntry* header_timeout_entry) {
   if (header_timeout_entry) {
```

Comment:
Looking at this it seems fine to pass through a null header to `trySetGlobalTimeout`?